File: web-file-manager-api/src/models/HostModel.ts
import os from 'os';

/**
 * Describes the host model.
 */
export default class HostModel {
    /**
     * The host name.
     */
    public readonly hostname: string;
    /**
     * The host platform.
     */
    public readonly platform: string;

    /**
     * Initializes the host model.
     */
    public constructor() {
        this.hostname = os.hostname();
        this.platform = os.platform();
    }
}

File: WebFileManagerApi/Models/HostModel.cs
using System;
using System.Runtime.InteropServices;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace WebFileManagerApi.Models
{
    /// <summary>
    /// This class is intended to represent the host model.
    /// </summary>
    public class HostModel
    {
        /// <summary>
        /// Gets the host name.
        /// </summary>
        public string Name => Environment.MachineName;
        /// <summary>
        /// Gets the host platform.
        /// </summary>
        public string Platform
        {
            get
            {
                if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
                {
                    return "Linux";
                }
                else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
                {
                    return "OSX";
                }
                else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                {
                    return "Windows";
                }
                else
                {
                    return "Unknown";
                }
            }
        }
        /// <summary>
        /// Gets the system time of the host.
        /// </summary>
        [JsonConverter(typeof(UnixDateTimeConverter))]
        public DateTime Time => DateTime.Now;
    }
}